pkg/config/config.go

```diff
@@ -37,38 +37,79 @@ var (
 	validRoles = []string{"controlplane", "node"}
 )
 
-type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+// Top level config
+type Config struct {
+	Components Components       `json:components`
+	Microshift MicroshiftConfig `json:microshift`
```

Comment:
It feels odd to imply that only a subset of the configuration file for MicroShift is for configuring MicroShift.

Maybe "logging" is a component? Then we could have something like

```
logging:
  level: 3
```

Can the MTU value go into the networking block?

And then can we move the "components" to each be top level entries? That would retain the nice organization of the per-component structs without introducing an extra level of nesting.

Reply:
@dinhxuanvu based on our discussion yesterday, I believe we said that MicroShift-specific params should not be gathered into a single region, but be "overlaid" onto the OpenShift config region it belongs to conceptually if that exists (e.g. "MTU" into networking, "bindAddress" into apiServer) and into the top level (or a new grouping) if not (e.g. log-level).

Reply:
> into the top level (or a new grouping) if not (e.g. log-level).

This is what I expected for logLevel.  I'm flexible for the containing type change, though `logging` seems to suffer from "what else would ever go into that region".  `microshift` doesn't offend my sensibilities, but it looks like a component to me.  `debugging` would be more general `logging`.

Reply:
The `debugging` struct sounds reasonable IMO. So I will construct it as:

```
type Debugging struct {
	LogLevel int `json:"logLevel"`
}
```